src/components/FilterButtons/FilterButtons.tsx

```diff
@@ -1,12 +1,16 @@
 import React, { useState } from "react";
-import PropTypes from "prop-types";
 import "./FilterButtons.css";
 
-function FilterButtons({ setFilterName }) {
-  const [selectedButton, setSelectedButton] = useState("Show all");
-  const handleClick = ({ target }) => {
-    setSelectedButton(target.textContent);
-    setFilterName(target.textContent);
+type FilterButtonsProps = { setFilterName: (value: string) => void };
+
+function FilterButtons({ setFilterName }: FilterButtonsProps): JSX.Element {
+  const [selectedButton, setSelectedButton] = useState<string>("Show all");
+
+  const handleClick = ({
+    currentTarget,
+  }: React.MouseEvent<HTMLButtonElement>): void => {
+    setSelectedButton(currentTarget.textContent!);
```

Comment:
È giusto usare in questo caso il Null assertion operator?

Reply:
no

Reply:
stai sempre dicendo a TS, "fidati, questo valore non può essere null"

Reply:
Le strade secondo me qui sono due:

1. lanci `setSelectedButton` solo se effettivamente textContent è presente
2. usi un [invariant](https://github.com/alexreardon/tiny-invariant) per fare throw nel caso textContent mancasse e poi lanci setSelectedButton

La strade 2 è più violenta, ma magari è giusta (dipende dal caso), perché non dovrebbe esistere un caso in cui textContent manchi (non lo so eh, sto facendo un ragionamento qui..)

Vedi tu!